client/control/components/Header.js

```diff
@@ -32,4 +52,56 @@ Header.propTypes = {
   currentLocation: pt.string.isRequired,
   routes: pt.array.isRequired,
   params: pt.object.isRequired,
+  token: pt.shape({ key: pt.string.isRequired }),
+  user: pt.object.isRequired,
 };
+
+export default connect(
+  (state, ownProps) => ({
+    ...ownProps,
+    token: state.auth.token,
+  }),
+)(Header);
+
+
+class APITokenExpander extends Component {
```

Comment:
In general, I almost never nest/declare multiple classes in the same file. Is this a component that may be used later, like in an admin panel or something? We may want to split this code up.